libraries/AP_Torqeedo/AP_Torqeedo.h

```diff
@@ -12,358 +12,116 @@
-   as described on the ArduPilot wiki. https://ardupilot.org/rover/docs/common-torqeedo.html
-   TQ Bus is a serial protocol over RS-485 meaning that a serial to RS-485 converter is required.
-
-       Tiller connection: Autopilot <-> Battery (master) <-> Motor
-       Motor connection:  Autopilot (master) <-> Motor
-
-    Communication between the components is always initiated by the master with replies sent within 25ms
-
-    Example "Remote (0x01)" reply message to allow tiller to control motor speed
-    Byte        Field Definition    Example Value   Comments
-    ---------------------------------------------------------------------------------
-    byte 0      Header              0xAC
-    byte 1      TargetAddress       0x00            see MsgAddress enum
-    byte 2      Message ID          0x00            only master populates this. replies have this set to zero
-    byte 3      Flags               0x05            bit0=pin present, bit2=motor speed valid
-    byte 4      Status              0x00            0x20 if byte3=4, 0x0 is byte3=5
-    byte 5      Motor Speed MSB     ----            Motor Speed MSB (-1000 to +1000)
-    byte 6      Motor Speed LSB     ----            Motor Speed LSB (-1000 to +1000)
-    byte 7      CRC-Maxim           ----            CRC-Maxim value
-    byte 8      Footer              0xAD
-
-   More details of the TQ Bus protocol are available from Torqeedo after signing an NDA.
- */
-
 #pragma once
 
 #include "AP_Torqeedo_config.h"
 
 #if HAL_TORQEEDO_ENABLED
 
-#include <AP_ESC_Telem/AP_ESC_Telem_Backend.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/Semaphores.h>
 #include <AP_Param/AP_Param.h>
+#include <GCS_MAVLink/GCS_MAVLink.h>
+
+#include "AP_Torqeedo_Params.h"
+
+// Maximum number of range finder instances available on this platform
+#define TORQEEDO_MAX_INSTANCES 3
+
+// first monitor is always the primary monitor
+#define TORQEEDO_PRIMARY_INSTANCE  0
+
+class AP_Torqeedo_Backend;
 
-#define TORQEEDO_MESSAGE_LEN_MAX    35  // messages are no more than 35 bytes
+class AP_Torqeedo 
+{
+    friend class AP_Torqeedo_Backend;
 
-class AP_Torqeedo : public AP_ESC_Telem_Backend {
 public:
     AP_Torqeedo();
 
+    // do not allow copies
     CLASS_NO_COPY(AP_Torqeedo);
 
-    static AP_Torqeedo* get_singleton();
+    // Torqeedo driver types
+    enum class Type {
+        None = 0,
 
-    // initialise driver
-    void init();
+#if AP_TORQEEDO_TQBUS_ENABLED
+        TQBus = 1,
+#endif
+    };
 
-    // consume incoming messages from motor, reply with latest motor speed
-    // runs in background thread
-    void thread_main();
+    enum class Status {
+        NotConnected = 0,
+        NoData = 1,
+        Good
+    };
 
-    // returns true if communicating with the motor
-    bool healthy();
+     // The TorqeedoState structure is filled in by the backend driver
+    struct Torqeedo_State { 
```

Comment:
If possible, let's remove this Torqeedo_State structure.  We can pass in the instance number to each backend's constructor.  The status can be maintained by the backend itself.  The var_info seems to be a duplilcate of backend_var_info isn't it?